### PR TITLE
_tier_preference docs changes for 7.16

### PR DESCRIPTION
**Pull request by joegallo**
Related to #76147, especially #79100

Adds document for the new `cluster.routing.allocation.enforce_default_tier_preference` setting, and references it from the manual (https://www.elastic.co/guide/en/elasticsearch/reference/current/migrate-index-allocation-filters.html) and automated (https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-migrate-to-data-tiers.html) instructions for moving from allocation filters to `_tier_preference`.

### CODE REVIEW

File: docs/reference/ilm/apis/migrate-to-data-tiers.asciidoc [hunk 1]
```diff
@@ -16,11 +16,12 @@ as indicated in the <<migrate-index-allocation-filters, Migrate index allocation
 filters to node roles>> page.
 
 This API provides an automated way of executing three out of the four manual steps listed
```
Comment (andreidan) [inline docs/reference/ilm/apis/migrate-to-data-tiers.asciidoc:18]: four out of five now? :)

**Review by andreidan — APPROVED** [review]
LGTM, thanks for documenting this

File: docs/reference/ilm/apis/migrate-to-data-tiers.asciidoc (hunk 1 above)
Comment (joegallo) [reply docs/reference/ilm/apis/migrate-to-data-tiers.asciidoc:18]: Ha! Quite! Good catch.

Comment (joegallo) [reply docs/reference/ilm/apis/migrate-to-data-tiers.asciidoc]: 4914e41

**Pull request merged**